Source/Core/Core/AchievementManager.cpp

```diff
@@ -1341,17 +1335,44 @@ void AchievementManager::ActivateDeactivateAchievement(AchievementId id, bool en
     rc_runtime_deactivate_achievement(&m_runtime, id);
 }
 
-void AchievementManager::GenerateRichPresence()
+void AchievementManager::DoPeriodically()
 {
-  Core::RunAsCPUThread([&] {
-    std::lock_guard lg{m_lock};
-    rc_runtime_get_richpresence(
-        &m_runtime, m_rich_presence.data(), RP_SIZE,
-        [](unsigned address, unsigned num_bytes, void* ud) {
-          return static_cast<AchievementManager*>(ud)->MemoryPeeker(address, num_bytes, ud);
-        },
-        this, nullptr);
-  });
+  if (!m_is_game_loaded)
+    return;
+  auto current_time = std::chrono::steady_clock::now();
+  if (current_time - m_last_rp_time > std::chrono::seconds{10})
+  {
+    if (Config::Get(Config::RA_RICH_PRESENCE_ENABLED))
+    {
+      Core::RunAsCPUThread([&] {
```

Comment:
#12658 is removing this function.  Maybe switch to `CPUThreadGuard`?

Reply:
The reason for its removal is that a braced scope containing a `CPUThreadGuard` is just as capable as the scope of the function passed to `RunAsCPUThread` (and easier to reason about, imo).